### PR TITLE
vdk-oracle: add multiple db connections and ingesters

### PR DESCRIPTION
**Pull request by duyguHsnHsn**
What: Add multiple db support for oracle plugin

Tests: functional test with two connections and two ingesters added

### CODE REVIEW

File: projects/vdk-plugins/vdk-oracle/README.md [hunk 1]
```diff
@@ -199,6 +199,81 @@ def run(job_input):
...
+oracle_password=reports_password
+oracle_host=localhost
+oracle_port=1523
+oracle_sid=FREE
+oracle_connection_string =localhost:1523/FREE
+oracle_thick_mode=False
+
+```
+
+You can specify which database to use in your data job by referencing the specific section name.
```
Comment (dakodakov) [inline projects/vdk-plugins/vdk-oracle/README.md:237]: Does it make sense to mention that  the section name, is going to be lowercased and used as the DB connection name?

File: projects/vdk-plugins/vdk-oracle/README.md [hunk 2]
```diff
@@ -199,6 +199,81 @@ def run(job_input):
...
+        method="oracle",
+        target="oracle"
+    )
+
+    # Ingest data into the reports database
+    payload_reports = {"col1": "value3", "col2": "value4"}
+    job_input.send_object_for_ingestion(
+        payload=payload_reports,
+        destination_table="reports_table",
+        method="oracle_reports",
```
[the remark this answers is not in the document]
Comment (duyguHsnHsn) [reply projects/vdk-plugins/vdk-oracle/README.md:272]: no, method should be always the name the connection/ingester was registered with

File: projects/vdk-plugins/vdk-oracle/README.md [hunk 3]
```diff
@@ -199,6 +199,81 @@ def run(job_input):
...
+oracle_password=password
+oracle_host=localhost
+oracle_port=1521
+oracle_sid=FREE
+oracle_connection_string =localhost:1521/FREE
+oracle_thick_mode=True
+
+[vdk_oracle_reports]
+oracle_user=reports_user
+oracle_password=reports_password
```
[the remark this answers is not in the document]
Comment (duyguHsnHsn) [reply projects/vdk-plugins/vdk-oracle/README.md:228]: I will add section about env variables and secrets to the RADME

File: projects/vdk-plugins/vdk-oracle/README.md [hunk 4]
```diff
@@ -199,6 +199,81 @@ def run(job_input):
     job_input.send_object_for_ingestion(payload=payload, destination_table="test_table")
 ```
 
+### Multiple Oracle Database
+
+#### Configuring Multiple Oracle Databases
+
+To manage multiple Oracle database connections within a data job,
+always configure the default database in the `[vdk]` section of the `config.ini` file.
+This section should contain the primary connection details that the application will use by default.
```
[the remark this answers is not in the document]
Comment (duyguHsnHsn) [reply projects/vdk-plugins/vdk-oracle/README.md:208]: ready

File: projects/vdk-plugins/vdk-oracle/src/vdk/plugin/oracle/oracle_plugin.py [hunk 5]
```diff
@@ -29,40 +29,64 @@
...
-                context.connections, conf.oracle_ingest_batch_size()
-            ),
-        )
+        for section in context.core_context.configuration.list_sections():
+            if section == "vdk":
+                connection_name = "oracle"  # the default database
+            else:
+                connection_name = section.lstrip("vdk_")
+
+            oracle_user, oracle_pass = conf.get_oracle_user(
```
[the remark this answers is not in the document]
Comment (duyguHsnHsn) [reply projects/vdk-plugins/vdk-oracle/src/vdk/plugin/oracle/oracle_plugin.py]: I added try/except block

File: projects/vdk-plugins/vdk-oracle/README.md (hunk 1 above)
Comment (duyguHsnHsn) [reply projects/vdk-plugins/vdk-oracle/README.md:242]: I added a note in the readme

File: projects/vdk-plugins/vdk-oracle/src/vdk/plugin/oracle/oracle_configuration.py [hunk 6]
```diff
@@ -22,45 +22,66 @@
 ORACLE_INGEST_BATCH_SIZE = "ORACLE_INGEST_BATCH_SIZE"
 
 
+def parse_boolean(value):
```
[the remark this answers is not in the document]
Comment (duyguHsnHsn) [reply projects/vdk-plugins/vdk-oracle/src/vdk/plugin/oracle/oracle_configuration.py:25]: Let's merge it like this. I will add the class in vdk-core. But vdk-core and plugins cannot be changed in one PR, I will open another PR after merging this one.

**Review by dakodakov — APPROVED** [review]
lgtm

**Pull request merged**